compiler/src/main/java/org/qbicc/type/PrimitiveType.java

```diff
@@ -0,0 +1,88 @@
+package org.qbicc.type;
+
+import java.util.function.Consumer;
+
+public class PrimitiveType {
```

Comment:
The idea here I think is really good and cleans up a lot of ugly stuff.  A couple minor notes though:

- The fact that this class ends in `...Type` and resides in the `.type` package makes it appear as if this is part of the type system, but it is not.  Maybe just call it `Primitive`?
- Would this be better done as an `enum`?

Reply:
Renamed `PrimitiveType` to `Primitive` and changed it to enum.